plutus-ledger-api/testlib/PlutusLedgerApi/Test/EvaluationContext.hs

```diff
@@ -1,29 +1,119 @@
 -- editorconfig-checker-disable-file
+{-# LANGUAGE TypeFamilies  #-}
+{-# LANGUAGE TypeOperators #-}
 module PlutusLedgerApi.Test.EvaluationContext
-    ( costModelParamsForTesting
-    , evalCtxForTesting
+    ( v1_costModelParamsForTesting
+    , v2_costModelParamsForTesting
+    , v3_costModelParamsForTesting
     ) where
 
-import PlutusCore.Evaluation.Machine.CostModelInterface as Plutus
-import PlutusCore.Evaluation.Machine.ExBudgetingDefaults qualified as Plutus
-import PlutusLedgerApi.Common
+import PlutusCore.Evaluation.Machine.BuiltinCostModel
+import PlutusCore.Evaluation.Machine.CostModelInterface
+import PlutusCore.Evaluation.Machine.ExBudgetingDefaults
+import PlutusCore.Evaluation.Machine.MachineParameters
 import PlutusPrelude
+import UntypedPlutusCore.Evaluation.Machine.Cek.CekMachineCosts
 
-import Data.Either.Extras
+import PlutusLedgerApi.Common as Common
+import PlutusLedgerApi.V1 qualified as V1
+import PlutusLedgerApi.V2 qualified as V2
+import PlutusLedgerApi.V3 qualified as V3
+
+import Barbies
+import Data.Aeson
+import Data.Functor.Identity
+import Data.Map qualified as Map
 import Data.Maybe
 
-{-| only for testing purposes: The default, raw cost model params.
+-- | Example values of costs for plutus-v3 , in expected ledger order, i.e. can be used in testing
+-- `V3.mkEvaluationContext . fmap snd`
+v3_costModelParamsForTesting :: [(V3.ParamName, Integer)]
+v3_costModelParamsForTesting = Map.toList $ fromJust $ extractCostModelParamsLedgerOrder v3_liftedCostModel
+
+-- | Example values of costs for plutus-v2, in expected ledger order, i.e. can be used in testing
+-- `V2.mkEvaluationContext . fmap snd`
+v2_costModelParamsForTesting :: [(V2.ParamName,Integer)]
+v2_costModelParamsForTesting = Map.toList $ fromJust $ extractCostModelParamsLedgerOrder v2_liftedCostModel
+
+-- | Example values of costs for plutus-v1, *in expected ledger order*, i.e. can be used in testing
+-- `V1.mkEvaluationContext . fmap snd`
+v1_costModelParamsForTesting :: [(V1.ParamName,Integer)]
+v1_costModelParamsForTesting = Map.toList $ fromJust $ extractCostModelParamsLedgerOrder v1_liftedCostModel
+
+v3_liftedCostModel, v2_liftedCostModel, v1_liftedCostModel
+    :: CostModel (CekMachineCostsBase Maybe)  (BuiltinCostModelBase MCostingFun)
+v3_liftedCostModel = liftToJust defaultCekCostModel
+v2_liftedCostModel = v3_liftedCostModel
+                    & machineCostModel
+                    %~ clearV3machineCostModel
+                    & builtinCostModel
+                    %~ clearV3builtinCostModel
+v1_liftedCostModel = v2_liftedCostModel
+                    & builtinCostModel
+                    %~ clearV2builtinCostModel
+
+-- Assign to `mempty` those builtins that the specific plutus version X introduces, and thus
+-- are missing from version X-1.
+-- Calling this function on a vX_BuiltinCostModel will turn it into a vX-1_BuiltinCostModel,
+-- by omitting the generation of costs concerning the missing builtins.
```

Comment:
This is kind of ugly. We already list the param names that we know about! Can we not do something with them?

Reply:
> This is kind of ugly. 
Yes it is certainly duplication.

>We already list the param names that we know about! Can we not do something with them?

You mean take advantage of this?

https://github.com/input-output-hk/plutus/blob/master/plutus-ledger-api/src/PlutusLedgerApi/Common/Versions.hs#L83-L114

Reply:
No, I mean this! https://github.com/input-output-hk/plutus/blob/master/plutus-ledger-api/src/PlutusLedgerApi/V1/ParamName.hs

Reply:
But then I have to write all the intercept and number of ModelThreeArguments, etc

Reply:
Hmmm, ok let me think about it a bit more

Reply:
Oh I see, those are for the flat version and this is the structured version... tricky :thinking: